### DISCUSSION TITLE
Related work

### DISCUSSION THREAD

**Discussion by nimrossum** [git-truck/git-truck#12] [Research]
# Collection of related work

## [RepoVis](https://ftp.isds.tugraz.at/pub/papers/feiner-vissoft2018-repovis.pdf)
This has two nice features. One is being the ability to change what the colors of elements mean. Where one could be that color represents the file type, another could be developer file dominance. Having a program with this feature could make it valuable in more cases during a project.

## [Visualizing file authorship and dominance](http://dspace.library.uvic.ca/bitstream/handle/1828/8419/Malik_Saad_MSc_2017.pdf?sequence=1&isAllowed=y)
Presents two metrics for measuring file dominance.

## [Polymetric Views-A Lightweight Visual Approach to Reverse Engineering](https://www.researchgate.net/publication/3188357_Polymetric_Views-A_Lightweight_Visual_Approach_to_Reverse_Engineering)
Another paper from the same project: https://www.inf.usi.ch/lanza/Downloads/Lanz04b.pdf

## [A Tool For Visualizing Line Oriented Software Statistics](http://www.cs.kent.edu/~jmaletic/cs63903/papers/Eick92.pdf)
[Image](https://user-images.githubusercontent.com/464519/153027776-14af5423-7887-4716-97fc-2b0669d70a62.png) of the visualization.

## [VISON: An Ontology-Based Approach for Software Visualization Tool Discoverability](https://arxiv.org/pdf/1908.04090.pdf)
This paper evaluates loads of software visualizers, might be a good source for inspiration.

**Comment by mircealungu** [comment]
https://www.researchgate.net/publication/4211302_How_developers_drive_software_evolution

**Comment by nimrossum** [comment]
Malik Saad: Git repository visualization: Visualizing file authorship and dominance
http://dspace.library.uvic.ca/bitstream/handle/1828/8419/Malik_Saad_MSc_2017.pdf?sequence=1&isAllowed=y

**Comment by nimrossum** [comment]
RepoVis: Visual Overviews and Full-Text Search
in Software Repositories
https://ftp.isds.tugraz.at/pub/papers/feiner-vissoft2018-repovis.pdf

RepoVis [1] have two nice features. One being the ability to change what the colors of elements mean. Where one could be that color represents the file type, another could be developer file dominance, as presented in different forms in [2]. Having a program with this feature could make it valuable in more cases during a project.

**Comment by emiljapelt** [comment]
This paper evaluates loads of software visualizers, might be a good source for inspiration
https://arxiv.org/pdf/1908.04090.pdf

**Comment by emiljapelt** [comment]
# Collection of related work

## [Visualizing a codebase ](https://next.github.com/projects/repo-visualization) 

## [Bus factor in practice](https://vovak.me/assets/papers/bus_factor_in_practice.pdf)

## [RepoVis](https://ftp.isds.tugraz.at/pub/papers/feiner-vissoft2018-repovis.pdf)
This has two nice features. One is being the ability to change what the colors of elements mean. Where one could be that color represents the file type, another could be developer file dominance. Having a program with this feature could make it valuable in more cases during a project.

## [Visualizing file authorship and dominance](http://dspace.library.uvic.ca/bitstream/handle/1828/8419/Malik_Saad_MSc_2017.pdf?sequence=1&isAllowed=y)
Presents two metrics for measuring file dominance.

## [Visually Analyzing the Structure and Code Quality of Component-based Web Applications](https://www.researchgate.net/profile/Hagen-Tarner/publication/354724873_Visually_Analyzing_the_Structure_and_Code_Quality_of_Component-based_Web_Applications/links/6149d2baa3df59440b9f495e/Visually-Analyzing-the-Structure-and-Code-Quality-of-Component-based-Web-Applications.pdf)

## [Polymetric Views-A Lightweight Visual Approach to Reverse Engineering](https://www.researchgate.net/publication/3188357_Polymetric_Views-A_Lightweight_Visual_Approach_to_Reverse_Engineering)
Another paper from the same project: https://www.inf.usi.ch/lanza/Downloads/Lanz04b.pdf

## [How developers drive software evolution](https://www.researchgate.net/publication/4211302_How_developers_drive_software_evolution)

## [A Tool For Visualizing Line Oriented Software Statistics](http://www.cs.kent.edu/~jmaletic/cs63903/papers/Eick92.pdf)
[Image](https://user-images.githubusercontent.com/464519/153027776-14af5423-7887-4716-97fc-2b0669d70a62.png) of the visualization.

## [VISON: An Ontology-Based Approach for Software Visualization Tool Discoverability](https://arxiv.org/pdf/1908.04090.pdf)
This paper evaluates loads of software visualizers, might be a good source for inspiration.

**Comment by mircealungu** [comment]
Today I had this idea of "git insight" being a good alternative name for the tool.
Turns out somebody stole the idea and already created a tool with that name: https://github.com/avimehenwal/git-insight. 
Admittedly his insights are less beautiful looking, but it's still related  work.

**Comment by mircealungu** [comment]
<img width="1890" alt="image" src="https://user-images.githubusercontent.com/464519/160190577-196b86e8-320c-40bd-8a14-d9cb2316389e.png">

this is a new tab that appeared in my visual studio code. 
seems that one has to pay 2usd/month for it... https://www.gitkraken.com/gitlens/plus-features

**Comment by mircealungu** [comment]
Very *artsy* related work in https://vis.cs.ucdavis.edu/papers/softvis_storylines.pdf

![image](https://user-images.githubusercontent.com/464519/160836845-56c0c7d3-975e-49b1-a37a-b63f10c6c451.png)
